packages/next/src/server/response-cache/index.ts

```diff
@@ -35,166 +51,130 @@ export default class ResponseCache {
       incrementalCache: IncrementalCache
     }
   ): Promise<ResponseCacheEntry | null> {
-    const { incrementalCache } = context
-    // ensure on-demand revalidate doesn't block normal requests
-    const pendingResponseKey = key
-      ? `${key}-${context.isOnDemandRevalidate ? '1' : '0'}`
-      : null
-
-    const pendingResponse = pendingResponseKey
-      ? this.pendingResponses.get(pendingResponseKey)
-      : null
+    // If there is no key for the cache, we can't possibly look this up in the
+    // cache so just return the result of the response generator.
+    if (!key) return responseGenerator(false, null)
 
-    if (pendingResponse) {
-      return pendingResponse
-    }
-
-    let resolver: (cacheEntry: ResponseCacheEntry | null) => void = () => {}
-    let rejecter: (error: Error) => void = () => {}
-    const promise: Promise<ResponseCacheEntry | null> = new Promise(
-      (resolve, reject) => {
-        resolver = resolve
-        rejecter = reject
-      }
-    )
-    if (pendingResponseKey) {
-      this.pendingResponses.set(pendingResponseKey, promise)
-    }
+    const { incrementalCache } = context
 
-    let resolved = false
-    const resolve = (cacheEntry: ResponseCacheEntry | null) => {
-      if (pendingResponseKey) {
-        // Ensure all reads from the cache get the latest value.
-        this.pendingResponses.set(
-          pendingResponseKey,
-          Promise.resolve(cacheEntry)
-        )
-      }
-      if (!resolved) {
-        resolved = true
-        resolver(cacheEntry)
-      }
-    }
+    return this.batcher.batch(
+      { key, isOnDemandRevalidate: false },
```

Comment:
should this be `context.isOnDemandRevalidate` rather than explicitly set to false?